CONTRIBUTING.md

```diff
@@ -47,3 +47,73 @@ mvn -f sdk/{projectForlderDir}/pom.xml -Dgpg.skip clean install
+
+This is going to be especially important after GA when releases aren't going to be the entire Azure SDK every time. If we're releasing a patch for a targeted azure-storage fix then we shouldn't need to build and release azure-core, we should be targeting the released versions and only building/releasing that update to azure-storage. It's worth noting that right now, in the version_client.txt, the dependency/current versions are the same. This will change once we GA, at which point the current version should be ahead of the dependency version.
+
+What about README files? Right now the README files in the repro end up getting into an odd state since things like samples and versions can get updated during the development process. We're in the process of versioning documentation with the releases which means that the docs are snapshot at the time of the release and then versioned and stored. This will allow the README files in the repro to have updated samples and versions that are setup for the next release.
+
+### Tooling, version files and marker tags
+All of the tooling lives under the eng\versioning directory.
+version_client.txt - Contains the Client library and versions
+version_data.txt - Contains Data library and versions
+update_versions.py - This is just a basic python script that will climb through the source tree and update POM and README files. The script utilizies tags within the files to do replacements and the tags are slightly different between the POM and README files.
+set_versions.py - This script should only be used by the build system when we start producing nightly ops builds. 
+
+In POM files this is done by inserting a specifically formatted comment on the same line as the version element.
+```xml
+  <groupId>MyGroup</groupId>
+  <artifactId>MyArtifact</artifactId>
+  <version>1.0.0-preview.1</version> <!-- {x-version-update;MyGroup:MyArtifact;[current|dependency]} -->
+```
+The last element of the tag would be current or dependency depending on the criteria previously explained.
+
+In README files this ends up being slightly different. Because the version tag is inside an XML element that we've explicitly telling a user to copy/paste into their product the comment tag really didn't make sense here. Instead there are tags before and after the XML element tags which effectively says "there's a version somewhere in between these two tags, when you find the line that matches replace it with the appropriate version of the group:artifact defined in the tag."
+
+\[//]: # ({x-version-update-start;MyGroup:MyArtifact;dependency})
```

Comment:
This definitely didn't correctly display in md.

Reply:
Updated. Will be updated with the next commit..